src/aequitas/flow/methods/preprocessing/unawareness.py

```diff
@@ -0,0 +1,168 @@
+from typing import Optional
+
+import pandas as pd
+import numpy as np
+from scipy.stats import chi2_contingency
+
+from ...utils import create_logger
+from .preprocessing import PreProcessing
+
+
+class Unawareness(PreProcessing):
+    def __init__(
+        self, top_k: Optional[int] = 1, correlation_threshold: Optional[float] = None
+    ):
+        """Removes features that are highly correlated with the sensitive attribute.
+
+        Parameters
+        ----------
+        top_k : int, optional
+            Number of features to remove. If None, the correlation_threshold
+            must be passed by the user. Defaults to 1.
+        correlation_threshold : float, optional
+            Features with a correlation value higher than this thresold are
+            removed. If None, the top_k parameter is used to determine how many
+            features to remove. Defaults to None.
+        """
+        self.logger = create_logger("methods.preprocessing.Unawareness")
+        self.logger.info("Instantiating an Unawareness preprocessing method.")
+        self.used_in_inference = False
```

Comment:
This one is used in inference. You remove the correlated features in the validation and test sets too!